common/Storyboarding/Util/OsbAnimationPool.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+
+namespace StorybrewCommon.Storyboarding.Util
+{
+    public class OsbAnimationPool : OsbSpritePool
+    {
+        private int frameCount;
+        private int frameDelay;
+        private OsbLoopType loopType;
+
+        public OsbAnimationPool(StoryboardLayer layer, string path, int frameCount, int frameDelay, OsbLoopType loopType, OsbOrigin origin, Action<OsbSprite, double, double> finalizeSprite = null) : base(layer, path, origin, finalizeSprite)
+        {
+            this.frameCount = frameCount;
+            this.frameDelay = frameDelay;
+            this.loopType = loopType;
+        }
+
+        public OsbAnimationPool(StoryboardLayer layer, string path, int frameCount, int frameDelay, OsbLoopType loopType, OsbOrigin origin, bool additive) : this(layer, path, frameCount, frameDelay, loopType, origin, additive ? (sprite, startTime, endTime) => sprite.Additive(startTime, endTime) : (Action<OsbSprite, double, double>)null)
+        {
+        }
+
+        public new OsbSprite Get(double startTime, double endTime)
+        {
+            var result = (PooledSprite)null;
+            foreach (var pooledSprite in pooledSprites)
+                if (pooledSprite.EndTime < startTime
+                    && startTime < pooledSprite.StartTime + MaxPoolDuration
+                    && (result == null || pooledSprite.StartTime < result.StartTime))
+                {
+                    result = pooledSprite;
+                }
+
+            if (result != null)
+            {
+                result.EndTime = endTime;
+                return result.Sprite;
+            }
+
+            var sprite = layer.CreateAnimation(path, frameCount, frameDelay, loopType, origin);
```

Comment:
Put the sprite creation in a virtual method and override it in this class instead of copying the whole method?
I'm not sure if there's a point to pooling animated sprites, since their animation starts on the earliest command's start time.